### PR TITLE
Add Prettier

### PR DESCRIPTION
**Pull request by lipis**
What do you think about prettifying some stuff with https://github.com/prettier/prettier?

Formatting all the support files in this project: `js`, `json`, `md`, `css`.

I also including the pre-commit hooks, so it will format stuff before making it to the repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,87 +11,97 @@ To install via Bower, simply do the following:
...
+| `fadeOutUp`       | `fadeOutUpBig`     | `flipInX`           | `flipInY`            |
+| `flipOutX`        | `flipOutY`         | `lightSpeedIn`      | `lightSpeedOut`      |
+| `rotateIn`        | `rotateInDownLeft` | `rotateInDownRight` | `rotateInUpLeft`     |
+| `rotateInUpRight` | `rotateOut`        | `rotateOutDownLeft` | `rotateOutDownRight` |
+| `rotateOutUpLeft` | `rotateOutUpRight` | `hinge`             | `jackInTheBox`       |
+| `rollIn`          | `rollOut`          | `zoomIn`            | `zoomInDown`         |
+| `zoomInLeft`      | `zoomInRight`      | `zoomInUp`          | `zoomOut`            |
+| `zoomOutDown`     | `zoomOutLeft`      | `zoomOutRight`      | `zoomOutUp`          |
+| `slideInDown`     | `slideInLeft`      | `slideInRight`      | `slideInUp`          |
+| `slideOutDown`    | `slideOutLeft`     | `slideOutRight`     | `slideOutUp`         |
```
Comment (lipis) [inline README.md]: This is impresses me every single time :)

File: package.json [hunk 2]
```diff
@@ -22,14 +22,31 @@
...
+  "lint-staged": {
+    "*.{js,json,md,css}": ["prettier --write", "git add"]
+  },
+  "prettier": {
+    "bracketSpacing": false,
+    "printWidth": 120,
+    "proseWrap": false,
+    "singleQuote": true,
+    "tabWidth": 2,
+    "trailingComma": "all"
```
Comment (lipis) [inline package.json]: Let me know if the rules are OK, or you want any changes. Here is the full list: https://prettier.io/docs/en/options.html

**Comment by eltonmesquita** [comment]
Thanks for the PR. I do like the idea of adding prettier to the project. I'll review the changes and merge it in the near future.

**Comment by lipis** [comment]
It's not that intrusive as the project was already pretty :) But this will make sure that everything is consistent..

**Comment by WarenGonzaga** [comment]
Alright thanks for your awesome suggestion for improving this project.

**Comment by lipis** [comment]
Prettier is cool :) Let's merge it.. 🎆 🍻 🍷

**Comment by eltonmesquita** [comment]
I really want to merge this, but we need a few things first:
- Right now Prettier has to run manually, probably via the code editor or CLI. Can you configure a Gulp task for it to run everytime you build the project?
- Can you solve the conflicts?
As soon as those issues are solved, I'm merging this.

**Comment by lipis** [comment]
sure.. I'll do it from the office :)

File: .editorconfig [hunk 3]
```diff
@@ -2,10 +2,11 @@
...
+end_of_line = lf
 indent_size = 2
+indent_style = space
+insert_final_newline = true
+max_line_length = 100
 tab_width = 2
-end_of_line = lf
-charset = utf-8
 trim_trailing_whitespace = true
-insert_final_newline = true
```
Comment (lipis) [inline .editorconfig]: added `max_line_length = 100` and sorted them alphabetically..

**Comment by lipis** [comment]
I resolved the conflicts and updated to the latest Prettier.

Also I'm not sure if we really need the gulp task, as I have installed the pre-commit hooks (wasn't working earlier) so whenever someone commits to the repo the changed files are being formatted with Prettier.

So on top of that you want to include one more task to prettify all files before building?! Shall I install the gulp-run for that?

**Comment by eltonmesquita** [comment]
Sorry for the huge delay.... But, no. No need to duplicate things, I hadn't seem that it was in a git hook.
Merging this.

**Comment by lipis** [comment]
Awesome!

**Comment by lipis** [comment]
You might want to run `npm run prettier` since most likely there are new files or maybe changes since the last time I did it..

**Comment by eltonmesquita** [comment]
Don't worry, I've committed some other changes and the hooks worked nicely! 😃

**Comment by WarenGonzaga** [comment]
Nice one...

**Pull request merged**